### ISSUE TITLE
How to set bind nic?

### ISSUE THREAD

**Issue by Some1OnLine** [joseantmazonsb/linguard#22]
Testing on a fresh hyper-v debian 11 guest vm.

I see how to set the port but what about the bind nic/adapter? Reason I ask is, It start and it looks like its running but on 127.0.0.1:8080 and I cant access via http://192.168.1.100:8080/. maybe im missing something. It's a headless vm with only ssh access.

either way im excited to play with this some more. looks promising.

Also found a type-o in install.sh on line 105 - replace requirements.txts with requirements.txt
pip3 install -r "${INSTALLATION_PATH}"/requirements.txt

Awesome work all around! 👍 

~SOL

**Comment by joseantmazonsb** [comment]
Hello there, thanks for opening the issue. 

I hadn't notice the typo in the install script and I will fix it as soon as possible. 

About the access issue, it should be enough to specify `host="0.0.0.0"` when running the Flask app:

```python

# web/server.py

...

    def run(self):
        logging.info(f"Running backend...")
        self.app.run(debug=self.args.debug, port=self.bindport, host="0.0.0.0")


server = Server()
server.run()
```

I think this should be the default behaviour or, at least, something you could specify through the settings, and I will also update the file asap.

Thanks for your kind words as well!

**Comment by Some1OnLine** [comment]
Thank you that did the trick. Im new to Flask/etc..But not new to being able to figure it out! Now I can get to the login as expected. At first login it asks to set a new password. I set the password and it logs in. But after login no changes I make get saved and after log out I can no longer log back in unless I delete the credentials.yaml which will re-prompt to to set a password and thus the loop starts over. 

Maybe a permission issue with files/folder? I trued setting 0777 as a test on all files and folders but didnt seem to help.

~SOL

**Comment by joseantmazonsb** [comment]
Hi again! I have fixed the login issues and added a way to specify the network adapter whose IP address will host the application. However, I could not get into the saving problem yet. I recently performed a big refactor to the configuration logic and I am afraid something broke along the way. I'll be looking into it in #23.

And one last thing... It would be much appreciated if you could install the branch #22 and validate that it solves all the problems you encountered (except #23) before I close the issue. You can manually checkout the branch or even rerun the install script specifying the branch as the last argument:

```bash
./install.sh path/to/install/folder "#22"
```

**Comment by Some1OnLine** [comment]
Thanks. Sure I will try/test as you referenced and let you know here in a few. Thanks again.

**Comment by Some1OnLine** [comment]
So far so good. I did as you said and it seems to be working better now both login and saving basic changes in webui.

I have a ton of existing configs for clients that im gonna try to import etc.. looks like webui doesnt let you set/view priv/pub keys. atm looks like I may need to manually build inside "linguard.sample.yaml"?

Ima play with it some more! Thanks again. 

~SOL

**Comment by joseantmazonsb** [comment]
Mmm, I think it would be a good thing to include some sort of import/export feature for interfaces and clients (clients can already be exported via Download button). I will work on that in #25.

About the keys, I initially decided not to show them for security reasons, but I think it's better to add a couple of password fields and let the user show/hide the keys. Scheduled in #26.

Thank you again for you kind words. I started this project for that very reason: I, myself, did not find any good Wireguard webguis. Some were too difficult to deploy, others just not powerful enough... So I decided to make my own and share it with the world.

All right, I will merge the PR linked to this issue and consider it resolved. Of course, feel free to open issues if you find some other bugs or have any feature requests :)
